test/langtools/tools/javac/annotations/parameter/ParameterAnnotations.java

```diff
@@ -0,0 +1,544 @@
+                    @Override
+                    public void accept(MethodBuilder builder, MethodElement element) {
+                        if (element instanceof RuntimeInvisibleParameterAnnotationsAttribute annos) {
+                            assert annos.parameterAnnotations().size() == 1;
+                            builder.accept(RuntimeInvisibleParameterAnnotationsAttribute.of(List.of(List.of(), annos.parameterAnnotations().get(0), List.of())));
+                        } else if (element instanceof RuntimeVisibleParameterAnnotationsAttribute annos) {
+                            assert annos.parameterAnnotations().size() == 1;
+                            builder.accept(RuntimeVisibleParameterAnnotationsAttribute.of(List.of(List.of(), annos.parameterAnnotations().get(0), List.of())));
+                        } else {
+                            builder.accept(element);
+                        }
+                    }
+                },
+               "long",
+               "- compiler.warn.runtime.invisible.parameter.annotations: T.class",
+               "1 warning");
+        //mismatched lengths on RuntimeVisibleParameterAnnotations and
+        //RuntimeInvisibleParameterAnnotations:
+        doTest(base,
+               """
+               import java.lang.annotation.*;
+               public class T {
+                   public T(@Visible @Invisible long l) {}
+               }
+               @Retention(RetentionPolicy.RUNTIME)
+               @interface Visible {}
+               @interface Invisible {}
+               """,
+               "T",
+                new MethodTransform() {
+                    @Override
+                    public void accept(MethodBuilder builder, MethodElement element) {
+                        if (element instanceof RuntimeInvisibleParameterAnnotationsAttribute annos) {
+                            assert annos.parameterAnnotations().size() == 1;
+                            builder.accept(annos); //keep intact
+                        } else if (element instanceof RuntimeVisibleParameterAnnotationsAttribute annos) {
+                            assert annos.parameterAnnotations().size() == 1;
+                            builder.accept(RuntimeVisibleParameterAnnotationsAttribute.of(List.of(List.of(), annos.parameterAnnotations().get(0), List.of())));
+                        } else {
+                            builder.accept(element);
+                        }
+                    }
+                },
+               "long",
+               "- compiler.warn.runtime.visible.invisible.param.annotations.mismatch: T.class",
+               "1 warning");
+    }
+
+    private MethodTransform NO_SIGNATURE = (builder, element) -> {
+        if (element instanceof SignatureAttribute) {
+            //ignore
+        } else {
+            builder.with(element);
+        }
+    };
+
+    private MethodTransform NO_SIGNATURE_NO_METHOD_PARAMETERS = (builder, element) -> {
```

Comment:
```suggestion
    private MethodTransform NO_SIGNATURE_NO_METHOD_PARAMETERS = MethodTransform.dropping(element -> element instanceof SignatureAttribute ||
            element instanceof MethodParametersAttribute);
```
Shortcuts like this.